subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -325,66 +323,104 @@ img_mgmt_state_confirm(void)
 	return rc;
 }
 
+#define REPORT_SLOT_ACTIVE	BIT(0)
+#define REPORT_SLOT_PENDING	BIT(1)
+#define REPORT_SLOT_CONFIRMED	BIT(2)
+#define REPORT_SLOT_PERMANENT	BIT(3)
+/* Return zcbor encoding result */
+static bool img_mgmt_state_encode_slot(zcbor_state_t *zse, uint32_t slot, int state_flags)
+{
+	uint32_t flags;
+	char vers_str[IMG_MGMT_VER_MAX_STR_LEN];
+	uint8_t hash[IMAGE_HASH_LEN]; /* SHA256 hash */
+	struct zcbor_string zhash = { .value = hash, .len = IMAGE_HASH_LEN };
+	struct image_version ver;
+	bool ok;
+	int rc = img_mgmt_read_info(slot, &ver, hash, &flags);
+
+	if (rc != 0) {
+		/* zcbor encoding did not fail */
+		return true;
+	}
+
+	ok = zcbor_map_start_encode(zse, MAX_IMG_CHARACTERISTICS)	&&
+	     (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 1	||
+	      (zcbor_tstr_put_lit(zse, "image")			&&
+	       zcbor_uint32_put(zse, slot >> 1)))			&&
+	     zcbor_tstr_put_lit(zse, "slot")				&&
+	     zcbor_uint32_put(zse, slot % 2)				&&
+	     zcbor_tstr_put_lit(zse, "version");
+
+	if (ok) {
+		if (img_mgmt_ver_str(&ver, vers_str) < 0) {
+			ok = zcbor_tstr_put_lit(zse, "<\?\?\?>");
+		} else {
+			vers_str[sizeof(vers_str) - 1] = '\0';
+			ok = zcbor_tstr_put_term(zse, vers_str);
+		}
+	}
+
+	ok = ok && zcbor_tstr_put_term(zse, "hash")						&&
+	     zcbor_bstr_encode(zse, &zhash)						&&
+	     ZCBOR_ENCODE_FLAG(zse, "bootable", !(flags & IMAGE_F_NON_BOOTABLE))	&&
+	     ZCBOR_ENCODE_FLAG(zse, "pending", state_flags & REPORT_SLOT_PENDING)	&&
+	     ZCBOR_ENCODE_FLAG(zse, "confirmed", state_flags & REPORT_SLOT_CONFIRMED)	&&
+	     ZCBOR_ENCODE_FLAG(zse, "active", state_flags & REPORT_SLOT_ACTIVE)		&&
+	     ZCBOR_ENCODE_FLAG(zse, "permanent", state_flags & REPORT_SLOT_PERMANENT)	&&
```

Comment:
Not that it needs to be changed but false fields can optionally be omitted

Reply:
`ZCBOR_ENCODE_FLAG` does that when `CONFIG_MCUMGR_GRP_IMG_FRUGAL_LIST` is set.